README.md

```diff
@@ -1,3 +1,139 @@
+
+
+Selectors returned by ReactSelector( selector ) are recognized as TestCafe selectors. You can combine them with regular selectors and filter with `.withText`, `.nth`, `.find` and [other](http://devexpress.github.io/testcafe/documentation/test-api/selecting-page-elements/selectors.html#functional-style-selectors) functions. To search for elements within a component, you can use the following combined approach.
+
+```js
+import ReactSelector from 'testcafe-react-selector';
+
+var itemsCount = ReactSelector('TodoApp').find('.items-count span');
+```
+
+Let’s use the API described above to add a task to a Todo list and check that the number of items changed.
+
+```js
+import ReactSelector from 'testcafe-react-selector';
+
+fixture `TODO list test`
+	.page('http://localhost:1337');
+
+test('Add new task', async t => {
+    const todoTextInput = ReactSelector('TodoInput');
+    const todoItem      = ReactSelector('TodoList TodoItem');
+
+    await t
+        .typeText(todoTextInput, 'My Item')
+        .pressKey('enter')
+        .expect(todoItem.count).eql(3);
+});
+```
+
+#### Obtaining component's props and state
+
+As an alternative to [testcafe snapshot properties](http://devexpress.github.io/testcafe/documentation/test-api/selecting-page-elements/dom-node-state.html), you can obtain `state` or `props` of a ReactJS component. You can use them in an assertion directly thus simplifying assertion logic.
+
+To obtain component properties and state, use the React selector’s `.getReact()` method.
+
+If you call this method with zero parameters, it returns an object of the following structure.
+```js
+{
+    props: <component_props>,
+    state: <component_state>
+}
+
+```
+
+Where `props` are React component properties excluding properties of its `children`, `state` – the state of the component.
+
+Example
+
+```js
+import ReactSelector from 'testcafe-react-selector';
+
+fixture `TODO list test`
+	.page('http://localhost:1337');
+
+test('Check list item', async t => {
+    const el = ReactSelector('TodoList');
+
+    await t.expect(el.getReact().props.priority).eql('High');
+    await t.expect(el.getReact().state.isActive).eql(false);
+});
+```
+
+As an alternative, the `.getReact()` method can take a function that returns the required property or state. This function acts as a filter. Its argument is an object returned by `.getReact()`, i.e. `{ props: ..., state: ...}`.
+```js
+ReactSelector('Component').getReact(({ props, state }) => {...})
+```
+
+Example
+
+```js
+import ReactSelector from 'testcafe-react-selector';
+
+fixture `TODO list test`
+    .page('http://localhost:1337');
+
+test('Check list item', async t => {
+    const el           = ReactSelector('TodoList');
+    const itemPriority = await  el.getReact(({ props }) => props.priority);
+    const isItemActive = await  el.getReact(({ state }) => state.isActive);
```

Comment:
Don't await for results, pass promise directly to assertion to enable smart assertion mechanism